lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -91,14 +100,46 @@ public byte[] Call(byte[] memento)
              
                 _dataReducer.Send(result);
             }
+            _stopped = true;
             return null;
         }
 
         /// <summary>
+        /// Received close event. 
+        /// If it is closed event sent from driver, set _break to true so that to stop the iteration in Call method
+        /// After _enforceCloseTimeout, if the task is still not closed, throw IMRUTaskSystemException to enforce stop. 
+        /// </summary>
+        /// <param name="closeEvent"></param>
+        public void OnNext(ICloseEvent closeEvent)
+        {
+            if (closeEvent.Value.IsPresent() && Encoding.UTF8.GetString(closeEvent.Value.Value).Equals(TaskManager.CloseTaskByDriver))
+            {
+                Logger.Log(Level.Info, "The task received close event with message: {0}.", Encoding.UTF8.GetString(closeEvent.Value.Value));
+                _break = true;
+            }
+            Thread.Sleep(TimeSpan.FromMilliseconds(_enforceCloseTimeoutMilliseconds));
```

Comment:
I won't say it is a hack. When driver ask a task to close, for whatever reason, if the task cannot close in the specified time frame, enforce close should happen. This is to make the code is robust. Even after we resolve that blocking call issue, we still need some back up plan to ensure sure the task is closed in case it cannot do it by itself.